src/components/elements/upload/UploaderBase.tsx

```diff
@@ -251,60 +161,191 @@ const Uploader: React.FC<UploaderProps> = ({
-    onDropAccepted: uploadAndCreate,
-    multiple: false,
+    multiple,
     accept,
     maxSize,
     noClick: true,
-    onDrop: (acceptedFiles, fileRejections) => {
-      fileRejections.forEach((file) => {
-        file.errors.forEach((err) => {
+    onDropAccepted: uploadAndCreate,
+    onDropRejected: (fileRejections) => {
+      const errors = fileRejections.flatMap((file) =>
+        file.errors.map((err) => {
+          const fileName = file.file.name;
           if (err.code === 'file-too-large') {
-            setError(
-              `Image is too large. File size must be under ${getReadableSize(
-                maxSize
-              )}`
-            );
+            return `${fileName} is too large. File size must be under ${getReadableSize(maxSize)}`;
           } else if (err.code === 'file-invalid-type') {
-            setError(
-              `Unsupported file type. ${
-                accept
-                  ? `Supported file types are: ${getFileTypesFromAccept(
-                      accept
-                    )}`
-                  : ''
-              }`
-            );
+            return `${fileName} is an unsupported file type. ${
+              accept
+                ? `Supported file types are: ${getFileTypesFromAccept(accept)}`
+                : ''
+            }`;
           } else {
-            setError(`Error: ${err.message || err.code}`);
+            return `${fileName} Error: ${err.message || err.code}`;
           }
-          setCurrentFile(undefined);
-        });
-      });
-      acceptedFiles.forEach(() => setError(undefined));
+        })
+      );
+      setErrors(errors);
     },
+    validator: uniqueNameValidator,
   });
 
-  const handleClear = useCallback<NonNullable<UploaderProps['onClear']>>(() => {
-    onClear(currentUpload, currentFile);
-    setCurrentFile(undefined);
-    setCurrentUpload(undefined);
-  }, [currentUpload, currentFile, onClear]);
+  const removeFile = useCallback(
+    (file: File | FileFieldsFragment) => {
+      const newFiles = currentFiles.filter((f) => f !== file);
+      const newUploads = currentUploads.filter(
+        // This is probably not ideal, but we do enforce file name uniqueness, so it should work
+        (upload) => upload.filename !== file.name
+      );
+      setCurrentFiles(newFiles);
+      setCurrentUploads(newUploads);
+      if (onUpload) onUpload(newUploads, newFiles);
+      if (onChange)
+        onChange([
+          ...existingFiles.filter((f) => f !== file),
+          ...newUploads.map((u) => u.blobId),
+        ]);
+    },
+    [currentFiles, currentUploads, existingFiles, onChange, onUpload]
+  );
 
-  return defaultChildren({
-    id,
-    file,
-    loading,
-    error,
-    previewUrl: fileImageUrl,
-    dragging: isDragActive,
-    clearFile: handleClear,
-    selectFile: open,
-    rootProps: getRootProps(),
-    inputProps: getInputProps(),
-    accept,
-    maxSize,
-  });
+  const showFileList = useMemo(
```

Comment:
Ok, even though I made my peace with the `defaultChildren` pattern and agree there's a time and place for it, here's my justification refactoring it out in this particular time and place: I started wanting to memoize things. I know that we could just make `defaultChildren` an actual react component, instead of a function but not a functional component(?). But since it is not actually being used anyway, I'm proposing to remove it. Happy to be persuaded otherwise, however!